### PR TITLE
[KOGITO-5942] Making possible to disable specific dashboard generation

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @r00ta  @jiripetrlik

See https://issues.redhat.com/browse/KOGITO-5942

Now there are two properties available (both defined inside GrafanaDashboardUtils):

`kogito.grafana.disabled.operational.dashboards`
`kogito.grafana.disabled.domain.dashboards`

Setting them inside application.properties (both Quarkus and Springboot) disable generation of specific dashboard, e.g.

`kogito.grafana.disabled.domain.dashboards=LoanEligibility`
`kogito.grafana.disabled.operational.dashboards=hello`

disable generation of domain dashboard `LoanEligibility` and operational dashboard `hello`.

The properties accept comma-separated list of values, to disable multiple dashboards.

Depends on 
https://github.com/kiegroup/kogito-examples/pull/991

### CODE REVIEW

File: kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/ProcessCodegen.java [hunk 1]
```diff
@@ -551,14 +551,24 @@ private static ChannelMappingStrategy getChannelMappingStrategy() {
 
         // generate Grafana dashboards
         if (context().getAddonsConfig().usePrometheusMonitoring()) {
-            String globalDbName = buildDashboardName(context().getGAV(), "Global");
-            String globalDbJson = generateOperationalDashboard(GLOBAL_OPERATIONAL_DASHBOARD_TEMPLATE, globalDbName, "Global", context().getGAV().orElse(KogitoGAV.EMPTY_GAV), false);
-            generatedFiles.addAll(DashboardGeneratedFileUtils.operational(globalDbJson, globalDbName + ".json"));
 
+            Optional<String> globalDbJson = generateOperationalDashboard(GLOBAL_OPERATIONAL_DASHBOARD_TEMPLATE,
```
Comment (danielezonca) [inline kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/ProcessCodegen.java:555]: Can you add a test in [`ProcessCodegenTest`](https://github.com/kiegroup/kogito-runtimes/blob/main/kogito-codegen-modules/kogito-codegen-processes/src/test/java/org/kie/kogito/codegen/process/ProcessCodegenTest.java)?

File: kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java [hunk 2]
```diff
@@ -338,17 +338,15 @@ private void generateRuleUnits(List<DroolsError> errors, List<GeneratedFile> gen
...
-            String dashboard = GrafanaConfigurationWriter.generateDomainSpecificDrlDashboard(
-                    domainDashboardDrlTemplate,
-                    dashboardName,
-                    ruleUnit.typeName(),
-                    context().getGAV().orElse(KogitoGAV.EMPTY_GAV),
-                    context().getAddonsConfig().useTracing());
-            generatedFiles.addAll(DashboardGeneratedFileUtils.domain(dashboard, dashboardName + ".json"));
-        }
-
+        Optional<String> domainDashboard = GrafanaConfigurationWriter.generateDomainSpecificDrlDashboard(
```
Comment (danielezonca) [inline kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java:341]: Can you add a test in [`IncrementalRuleCodegenTest`](https://github.com/kiegroup/kogito-runtimes/blob/main/kogito-codegen-modules/kogito-codegen-rules/src/test/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegenTest.java)?

**Review by r00ta — COMMENTED** [review]
Thank you @gitgabrio ! Small comment from my side

**Review by jiripetrlik — APPROVED** [review]

File: kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/context/KogitoBuildContext.java [hunk 3]
```diff
@@ -145,6 +145,8 @@ default boolean isValidationSupported() {
 
     Optional<KogitoGAV> getGAV();
 
+    Properties getProperties();
```
[the remark this answers is not in the document]
Comment (gitgabrio) [reply kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/context/KogitoBuildContext.java:148]: @danielezonca 
Done - replaced with Map

File: kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/ProcessCodegen.java (hunk 1 above)
Comment (gitgabrio) [reply kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/ProcessCodegen.java:555]: @danielezonca 
Done (also inside IncrementalRuleCodegenTest)

File: kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java (hunk 2 above)
Comment (gitgabrio) [reply kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java:341]: @danielezonca 
Done

File: grafana-api/src/main/java/org/kie/kogito/grafana/GrafanaConfigurationWriter.java [hunk 4]
```diff
@@ -50,76 +54,99 @@ private GrafanaConfigurationWriter() {
...
-    public static String generateDomainSpecificDrlDashboard(String templatePath, String dashboardName, String endpoint, KogitoGAV gav, boolean generateAuditLink) {
-        String template = readStandardDashboard(templatePath);
-        template = customizeTemplate(template, endpoint, gav.getArtifactId(), gav.getVersion());
-
-        JGrafana jgrafana = initialize(template, String.format("%s - Domain Dashboard", dashboardName), generateAuditLink);
-
-        return serialize(jgrafana);
+    public static Optional<String> generateDomainSpecificDrlDashboard(String templatePath, String dashboardIdentifier, Properties properties, String endpoint, KogitoGAV gav,
+            boolean generateAuditLink) {
+        if (isDomainDashboardEnabled(properties, dashboardIdentifier)) {
```
[the remark this answers is not in the document]
Comment (gitgabrio) [reply grafana-api/src/main/java/org/kie/kogito/grafana/GrafanaConfigurationWriter.java:140]: @r00ta 
done

**Review by danielezonca — APPROVED** [review]

**Review by r00ta — APPROVED** [review]

**Pull request merged**